src/app/components/core/configuration-manager/configuration-group/configuration-group.component.ts

```diff
@@ -0,0 +1,181 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { ConfigurationControlService, ConfigurationService, RolesService } from '../../../../services';
+import { DeveloperFeaturesService } from '../../../../services/developer-features.service';
+import { chain, cloneDeep } from 'lodash';
+
+@Component({
+  selector: 'app-configuration-group',
+  templateUrl: './configuration-group.component.html',
+  styleUrls: ['./configuration-group.component.css']
+})
+export class ConfigurationGroupComponent implements OnInit {
+
+  selectedGroup = 'Default Configuration';
+  @Input() category;
+  groups = [];
+
+  @Output() changedConfigEvent = new EventEmitter<any>();
+  @Output() formStatusEvent = new EventEmitter<boolean>();
+  @Output() changedAdvanceConfigEvent = new EventEmitter<any>();
+
+  // To hold the changed configuration values of a plugin
+  configFormValues = {};
+
+  pages = ['south', 'north', 'notification'];
+  @Input() from;
+  categoryKey = '';
+
+  advanceConfiguration: any
+  securityConfiguration: any;
+  changedAdvanceConfiguration: any;
+  changedSecurityConfiguration: any;
+
+  constructor(
+    public developerFeaturesService: DeveloperFeaturesService,
+    public rolesService: RolesService,
+    private configService: ConfigurationService,
+    private configurationControlService: ConfigurationControlService,
+  ) { }
+
+  ngOnInit() { }
+
+  ngOnChanges() {
+    this.categeryConfiguration();
+    this.getChildConfigData();
+  }
+
+  public updateCategroyConfig(config) {
+    this.category.config = config;
+    this.getChildConfigData();
+    this.categeryConfiguration();
+  }
+
+  categeryConfiguration() {
+    this.groups = [];
+    const configItems = Object.keys(this.category.config).map(k => {
+      this.category.config[k].key = k;
+      return this.category.config[k];
+    });
+
+    this.groups = chain(configItems).groupBy(x => x.group).map((v, k) => {
+      if (k != "undefined") {
+        return { category: this.category.name, group: k, config: Object.assign({}, ...v.map(vl => { return { [vl.key]: vl } })) }
+      } else {
+        return { category: this.category.name, group: "Default Configuration", config: Object.assign({}, ...v.map(vl => { return { [vl.key]: vl } })) }
+      }
+    }).value();
+  }
+
+  /**
+   * Set configuration of the selected child category
+   * @param category Object{key, description, displayName}
+   */
+  selectTab(tab: string) {
+    if (tab !== this.selectedGroup) {
+      this.selectedGroup = tab;
+    }
+  }
+
+  public getChildConfigData() {
+    // No advance configuration on add form
+    if (this.pages.includes(this.from) && this.category) {
+      this.categoryKey = this.category.name;
+      this.checkIfAdvanceConfig(this.category.name)
```

Comment:
> No advance configuration on add form

then why this.checkIfAdvanceConfig

Reply:
> this.pages.includes(this.from)
This condition is to check for modal window, means this.checkIfAdvanceConfig will be called on modal window only.